sklearn/metrics/classification.py

```diff
@@ -42,6 +42,17 @@
 from ..exceptions import UndefinedMetricWarning
 
 
+def _check_zero_division(zero_division):
+    if isinstance(zero_division, str):
+        if zero_division != "warn":
+            raise ValueError('zero_division must be one of ["warn", 0, 1]')
+    elif isinstance(zero_division, (int, float)):
+        if zero_division not in [0, 1]:
+            raise ValueError('zero_division must be one of ["warn", 0, 1]')
+    else:
+        raise TypeError('zero_division must be one of ["warn", 0, 1]')
```

Comment:
we aren't usually pedantic like this in distinguishing TypeErrors for parameter types frrom `ValueErrors`

Reply:
simplified in the new version